libraries/chain_kv/include/b1/session/shared_bytes.hpp

```diff
@@ -114,10 +107,11 @@ inline shared_bytes shared_bytes::operator++(int) {
       buffer.pop_back();
    }
 
-   return eosio::session::make_shared_bytes(buffer.data(), buffer.size());
+   return eosio::session::shared_bytes(buffer.data(), buffer.size());
 }
```

Comment:
`operator++(int)` 
- Seems like this should be a method and not an operator++ as it does not modify this. I expect an operator++(int) to modify this and return the old value. If it was an operator++ then it should be pre increment since it returns a new value not the current value.
- It looks like `if (++buffer.back())` is to detect overflow and not to detect a value of -1. Either add check for -1 or verify not max char as overflow of signed is undefined.
- This method could be optimized. It makes two copies of bytes. One into temp buffer and another into the return value. It should only make one copy at most.